src/time.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::{Instant, SystemTime};
+
+#[derive(Clone)]
+pub struct Converter {
+    now: Instant,
+    sys_now: SystemTime,
+}
+
+impl Converter {
+    pub fn new() -> Self {
+        Self {
+            sys_now: SystemTime::now(),
```

Comment:
The now and sys_now values can be arbitrary, they only have o represent the ~same moment in time so that we can convert. One of those is always subtracted and the other added, so they always cancel each other.